frontend/src/lib/components/apps/editor/settingsPanel/MoveToOtherGrid.svelte

```diff
@@ -0,0 +1,99 @@
+<script lang="ts">
+	import { Button } from '$lib/components/common'
+	import { dirtyStore } from '$lib/components/common/confirmationModal/dirtyStore'
+	import { faCopy } from '@fortawesome/free-solid-svg-icons'
+	import { getContext } from 'svelte'
+	import type { App, AppEditorContext } from '../../types'
+	import { createNewGridItem, deleteGridItem, findGridItem, insertNewGridItem } from '../appUtils'
+	import type { AppComponent } from '../component'
+
+	export let component: AppComponent | undefined
+	export let parent: string | undefined
+
+	const { app, selectedComponent } = getContext<AppEditorContext>('AppEditorContext')
+
+	function listAllSubGrids(app: App) {
+		return app.subgrids ? Object.keys(app.subgrids) : []
+	}
+
+	function moveComponent(id: string, targetId: string, targetSubGridIndex: number) {
+		const node = findGridItem($app, id)
+
+		if (!node) {
+			return
+		}
+
+		$selectedComponent = undefined
+
+		const data: AppComponent = JSON.parse(JSON.stringify(node.data))
+		$dirtyStore = true
+
+		deleteGridItem($app, data, parent)
+
+		insertNewGridItem(
+			$app,
+			data,
+			{
+				parentComponentId: targetId,
+				subGridIndex: targetSubGridIndex
+			},
+			true
+		)
+
+		$selectedComponent = id
+	}
+
+	$: availableGrids = listAllSubGrids($app)
+
+	function onMove() {
+		if (component) {
+			const [targetId, targetSubGridIndex] = (
+				document.getElementById('move-to-grid') as HTMLSelectElement
```

Comment:
why not bind the value of the select ?

Reply:
https://svelte.dev/tutorial/select-bindings

Reply:
Done!